### PR TITLE
bootstrap.sh: Better doc pruning / support acdir dirlists

### PR DESCRIPTION
**Pull request by felixbuenemann**
* Clean doc dir with find instead of moving files to keep to temp dir (old code could break on ctrl+c)
* Limited support for automake ACDIR dirlists (require for mac homebrew, no wildcard handling)

### CODE REVIEW

File: bootstrap.sh [hunk 1]
```diff
@@ -13,47 +13,41 @@ rm -f swig/vipsCC/*.cxx
...
 mkdir -p m4
-cp $ACDIR/codeset.m4 m4
-cp $ACDIR/gettext.m4 m4
-cp $ACDIR/glibc21.m4 m4
-cp $ACDIR/iconv.m4 m4
-cp $ACDIR/lcmessage.m4 m4
-cp $ACDIR/progtest.m4 m4
-cp $ACDIR/introspection.m4 m4
+# glib-gettextize asks us to copy these files to m4 if they aren't there:
+files="codeset gettext glibc21 iconv isc-posix lcmessage progtest introspection"
```
Comment (felixbuenemann) [inline bootstrap.sh]: I have none of these files except introspection.m4 in /usr/local/share/aclocal and doc generation seems to work fine.
Is this actually required on linux?

**Comment by felixbuenemann** [comment]
Seems I need to adapt the `find` command for GNU find (works fine on BSD find):

```
find: warning: you have specified the -depth option after a non-option argument (, but options are not positional (-depth affects tests specified before it as well as those specified after it).  Please specify options before other arguments.
```

**Comment by felixbuenemann** [comment]
Adjusted to work with both GNU and BSD find.

File: bootstrap.sh [hunk 2]
```diff
@@ -13,47 +13,43 @@ rm -f swig/vipsCC/*.cxx
...
-	mv doc/*.py poop ; \
-	rm -rf doc/* ; \
-	mv poop/* doc ; \
-	rmdir poop \
-)
+find doc -depth \( \
+      -path doc/libvips-docs.xml.in \
+   -o -path doc/Makefile.am \
+   -o -path 'doc/images/*' \
+   -o -name '*.xml' ! -name libvips-docs.xml ! -path 'doc/xml/*' \
```
Comment (felixbuenemann) [inline bootstrap.sh]: The generated libvips-docs.xml was previously preserved, because it matched `*.xml`.

**Comment by jcupitt** [comment]
I had a good idea: how about using git to remove all files not under source control from `doc/`? That would be much simpler and safer than a find command or that craziness with `mv` I had. 

Googling seems to suggest something like:

```
git clean -dfX doc
```

should do it. It seems `libvips-docs.xml` is under source control, it should probably be added to `.gitignore`.

**Comment by felixbuenemann** [comment]
That was my initial idea, I thought you intentionally avoided git so it also works with a tarball.

**Comment by jcupitt** [comment]
bootstrap predates git, vips was SVN back when it was made (!!).

You only need bootstrap when building from source control. Tarballs have configure all set up and ready to go. I think it's OK to depend on git within bootstrap.

**Comment by felixbuenemann** [comment]
The gtk-doc.make should probably also be removed from git, it's auto-generated by `gtkdocize`.

**Comment by felixbuenemann** [comment]
Oh and I have fond memories of the times of svn and slandering into the kitchen to brew a cup of coffee and have a chat while waiting for `svn checkout` to complete… ;-)

**Comment by jcupitt** [comment]
I've tried removing `gtk-doc.make` before, but gtkdocize on OS X wouldn't run without that file, so you couldn't then build from git on OS X. Perhaps that's been fixed.

**Comment by felixbuenemann** [comment]
Seems to be fixed, it gets generated on my machine (OS X 10.11) with `gtkdocize (gtk-doc) 1.24`.

**Comment by jcupitt** [comment]
Merged and I'll have a go with git clean in a separate commit. Thanks!

**Comment by jcupitt** [comment]
Actually, you're right, let's not add another dependency. find it is.

**Pull request merged**